test/rules/parameter_assignments.dart

```diff
@@ -45,3 +49,34 @@ void otherBad1(int parameter) { // LINT
   parameter += 3;
```

Comment:
I thought we decided to lint at the mutating node when we know it, of course we could proceed to find it but that will make rule more complex, specially for expressions like `variable++`. I would say that if finding the offending node is hard, is a signal that the method/function is doing too much, but that's how I think about my code, I guess it might be a bit harsh expectation for all users. What do you think?